content/blog/super-charging-our-github-action-with-the-pulumi-automation-api/index.md

```diff
@@ -0,0 +1,200 @@
+---
+title: "Supercharging our GitHub Action with the Automation API"
+authors: 
+  - paul-stack
+#  - simen-olsen
+tags: 
+  - GitHub Actions
+  - automation api
+date: "2021-02-24"
+#meta_image: ""
+meta_desc: "Simen Olsen teams up with Paul Stack to talk about the new GitHub Action that has been super charged with the Pulumi Automation API"
+---
+
+**Guest Article:** <About Simen here>
+
+In this article, we'll take a look at how our [GitHub Action](https://github.com/pulumi/actions) has been re-architected to be built on top of our [Automation API]({{< relref "/blog/automation-api" >}}).
+
+<!--more-->
+
+Pulumi was a GitHub Actions launch partner in 2018, at which time we launched our action using the original [Docker-based GitHub Action type](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action).
+This approach allowed us to package up all the necessary pre-requisities and the Pulumi CLI to allow users to
+be able to run Pulumi commands. As the Pulumi ecosystem has grown, so too did the container that GitHub Actions run.
+
+Since then, an alternative way to create a GitHub Action has become available,
+[by simply writing JavaScript](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action).
+At Pulumi, and in our community, we love expressing everything we possibly can in code, so this new support got us quite
+excited about the "better together" possibilities! [Enter Pulumi's new Automation API](https://www.pulumi.com/blog/automation-api/),
+a new feature we launched in public preview in October 2020. This new capability lets you do infrastructure as code
+&mdash; in pure code &mdash; without even needing to run a CLI. This sounded like a perfect match for JavaScript Actions.
+It allows us to programmatically invoke the right Pulumi commands in response to actions, without needing to awkwardly
+script a CLI. Instead, we were able to leverage the Automation API to fundamentally rethink our GitHub Action.
```

Comment:
```suggestion
Since then, GitHub released an alternative way to create a GitHub Action
[by simply writing JavaScript](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action).
At Pulumi, and in our community, we love expressing everything we possibly can in code, so this new support got us quite
excited about the "better together" possibilities! [Enter Pulumi's new Automation API](https://www.pulumi.com/blog/automation-api/),
a new feature we launched in public preview in October 2020. This new capability lets you do infrastructure as code
&mdash;in pure code&mdash; without even needing to run a CLI. This sounded like a perfect match for a JavaScript Action.
Creating a JavaScript Action allows us to programmatically invoke the right Pulumi commands in response to actions, without needing to awkwardly
script a CLI. Instead, we were able to leverage the Automation API to fundamentally rethink our GitHub Action.
```